### PR TITLE
Remove document types that don't exist anymore

### PR DESCRIPTION
**Pull request by tijmenb**
These document types have been removed from the system. Found by running `ContentItem.distinct(:document_type)` in the content store in staging.

https://trello.com/c/Jm6K9Tpx

### CODE REVIEW

File: lib/govuk_content_schemas/allowed_document_types.yml [hunk 1]
```diff
@@ -149,9 +138,7 @@
 - travel_advice_index
 - unpublishing
 - utaac_decision
-- vanish
```
Comment (cbaines) [inline lib/govuk_content_schemas/allowed_document_types.yml]: As far as I know, `vanish` is still being used in the Publishing API. I haven't checked any of the other document types yet.

Comment (tijmenb) [reply lib/govuk_content_schemas/allowed_document_types.yml]: Thanks, updated.

File: dist/formats/generic/frontend/schema.json [hunk 2]
```diff
@@ -164,7 +156,6 @@
         "smart_answer",
         "social_media_use",
         "special_route",
-        "specialist_document",
```
Comment (nickcolley) [inline dist/formats/generic/frontend/schema.json]: these definitely are a thing, what document type covers our specialist documents?

Comment (tijmenb) [reply dist/formats/generic/frontend/schema.json]: `specialist_document` is the `schema_name`, not the `document_type`. Different types of specialist docs have different document types: `aaib_report`, `maib_report`, `service_standard_report`, etc.

Comment (nickcolley) [reply dist/formats/generic/frontend/schema.json]: that's always confused me, thanks

File: lib/govuk_content_schemas/allowed_document_types.yml [hunk 3]
```diff
@@ -35,16 +33,12 @@
 - esi_fund
 - fatality_notice
 - field_of_operation
-- financial_release
-- financial_releases_campaign
 - financial_releases_geoblocker
```
Comment (steventux) [inline lib/govuk_content_schemas/allowed_document_types.yml]: It's a shame this and the `financial_releases_success` document type aren't gone from the content store, they don't appear to do geoblocking any more https://www.gov.uk/lloydsshares/lloyds-share-offer-press-releases/location.

Comment (tijmenb) [reply lib/govuk_content_schemas/allowed_document_types.yml]: Yeah, it's on my list of things to fix next.

**Pull request merged**